### PR TITLE
fix(list): throw error if allocation is invalid

### LINKED ISSUE #246: Listing a file in another allocation does not fail but returns no output
Users should not be able to list files in an allocation they do not own.
While it is true they cannot do this, when calling zbox list, a successful response is returned with no files listed.
image
If the allocation does not exist an error is returned:
image

The issue may actually lie with the REST API rather than the CLI.

2 issues with this:
1. Returning a blank response leaks data to an attacker by letting them know that an allocation exists 
2. Returning a zero exist code breaks any scripts they may be using to determine a success/failure 

Steps to reproduce:
#ORIGINAL WALLET
./zbox register
./zwallet faucet --methodName pour --tokens 1.000000 --input {}
./zbox newallocation --expire 1h --size 10000 --lock 0.5
./zbox upload --allocation <> --localpath /var/folders/vy/j3qpwd9s2dvbdvd_g5dhwvf80000gn/T/HY2PDunsiH_test.txt --remotepath /HY2PDunsiH_test.txt

#NEW WALLET
./zbox register
./zbox list --allocation <> --json  --remotepath /

### PR DESCRIPTION
**Pull request by cnlangzi**
### Changes
-

### Fixes
- fixed: https://github.com/0chain/zboxcli/issues/246

### Tests
Tasks to complete before merging PR:
- [ ]  Ensure system tests are passing. If not [Run them manually](https://github.com/0chain/gosdk/actions/workflows/system_tests.yml) to check for any regressions :clipboard:
- [ ]  Do any new system tests need added to test this change? do any existing system tests need updated? If so create a PR at [0chain/system_test](https://github.com/0chain/system_test)
- [ ]  Merge your system tests PR to master AFTER merging this PR

### Associated PRs (Link as appropriate):
- system_test: https://github.com/0chain/system_test/pull/452
- zboxcli: https://github.com/0chain/zboxcli/pull/310

### CODE REVIEW

**Review by peterlimg — APPROVED** [review]

File: zboxcore/sdk/listworker.go [hunk 1]
```diff
@@ -143,15 +143,22 @@ func (req *ListRequest) getlistFromBlobbers() []*listResponse {
...
+	var err error
+	var errNum int
 	for i := 0; i < len(lR); i++ {
 		req.consensus = 0
 		ti := lR[i]
-		if ti.err != nil || ti.ref == nil {
+		if ti.err != nil {
+			err = ti.err
+			errNum++
+			continue
```
Comment (boddumanohar) [inline zboxcore/sdk/listworker.go:159]: if we see the error, why not return error there itself?

Comment (cnlangzi) [reply zboxcore/sdk/listworker.go:159]: @boddumanohar  list should work even 1 blobber fails.

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | master  |
| gosdk  | fix/throws_error_on_list  |
| zboxcli  | fix/throws_error_on_list  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [cancelled] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | chore/upraded_deps  |
| gosdk  | fix/throws_error_on_list  |
| zboxcli  | fix/throws_error_on_list  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [cancelled] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | master  |
| gosdk  | fix/throws_error_on_list  |
| zboxcli  | fix/throws_error_on_list  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | fix/throws_error_on_list  |
| gosdk  | fix/throws_error_on_list  |
| zboxcli  | fix/throws_error_on_list  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [cancelled] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | fix/throws_error_on_list  |
| gosdk  | fix/throws_error_on_list  |
| zboxcli  | fix/throws_error_on_list  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [cancelled] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | fix/throws_error_on_list  |
| gosdk  | fix/throws_error_on_list  |
| zboxcli  | fix/throws_error_on_list  |
| zwalletcli  | fix/throws_error_on_list  |
| 0chain  | staging  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [success] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | fix/throws_error_on_list  |
| gosdk  | fix/throws_error_on_list  |
| zboxcli  | fix/throws_error_on_list  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Pull request merged**